extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -14,6 +17,103 @@ import {
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Validates string length of shopping config attributes
+ * @param {?string} field
+ * @param {?string} value
```

Comment:
Sounds good, thanks for the insight, Corey. I'll keep this in mind when having future optional parameters.

Reply:
After writing my comment, I realized that `null` can be used as a [valid value](https://www.w3schools.com/js/js_json_datatypes.asp) within JSON. This means that it is possible for a well-intentioned publisher to manually set `null` for an optional key (e.g., `'productColor': null`) instead of letting the value be `undefined` by simply omitting the key.

Therefore, I think that the value annotations in the validation methods should actually all be `@param {?string=} value/str/number/url` to signify that both the key's value could possibly be `null` or `undefined`. Additionally, you'll need to update your validation functions to also check for null values instead of just undefined ones

Reply:
Sounds good, this is what I had at the beginning, as == checks for both null and undefined.
https://github.com/ampproject/amphtml/pull/37474#discussion_r796934950